### ISSUE TITLE
Problem with numba==0.54.0

### ISSUE THREAD

**Issue by windisch** [shap/shap#2149]
When running

```from shap import KernelExplainer```

using `shap==0.39.0` together with `numba==0.54.0`,  then the following exception is raised when `numpy<1.20.0` is used (for `numpy==1.20.0` everything seems fine). 

This may be a numpy related issue, but you may could bin the numba and numpy requirements in shap to early detect version conflicts?

```
/usr/local/lib/python3.7/dist-packages/shap/__init__.py in <module>
     10     warnings.warn("As of version 0.29.0 shap only supports Python 3 (not 2)!")
     11
---> 12 from ._explanation import Explanation, Cohorts
     13
     14 # explainers

/usr/local/lib/python3.7/dist-packages/shap/_explanation.py in <module>
     10 from slicer import Slicer, Alias, Obj
     11 # from ._order import Order
---> 12 from .utils._general import OpChain
     13
     14 # slicer confuses pylint...

/usr/local/lib/python3.7/dist-packages/shap/utils/__init__.py in <module>
----> 1 from ._clustering import hclust_ordering, partition_tree, partition_tree_shuffle, delta_minimization_order, hclust
      2 from ._general import approximate_interactions, potential_interactions, sample, safe_isinstance, assert_import, record_import_error
      3 from ._general import shapley_coefficients, convert_name, format_value, ordinal_str, OpChain
      4 from ._show_progress import show_progress
      5 from ._masked_model import MaskedModel, make_masks

/usr/local/lib/python3.7/dist-packages/shap/utils/_clustering.py in <module>
      2 import scipy as sp
      3 from scipy.spatial.distance import pdist
----> 4 from numba import jit
      5 import sklearn
      6 import warnings

/usr/local/lib/python3.7/dist-packages/numba/__init__.py in <module>
     37
     38 # Re-export decorators
---> 39 from numba.core.decorators import (cfunc, generated_jit, jit, njit, stencil,
     40                                    jit_module)
     41

/usr/local/lib/python3.7/dist-packages/numba/core/decorators.py in <module>
     10
     11 from numba.core.errors import DeprecationError, NumbaDeprecationWarning
---> 12 from numba.stencils.stencil import stencil
     13 from numba.core import config, extending, sigutils, registry
     14

/usr/local/lib/python3.7/dist-packages/numba/stencils/stencil.py in <module>
      9 from llvmlite import ir as lir
     10
---> 11 from numba.core import types, typing, utils, ir, config, ir_utils, registry
     12 from numba.core.typing.templates import (CallableTemplate, signature,
     13                                          infer_global, AbstractTemplate)

/usr/local/lib/python3.7/dist-packages/numba/core/ir_utils.py in <module>
     14
     15 import numba
---> 16 from numba.core.extending import _Intrinsic
     17 from numba.core import types, utils, typing, ir, analysis, postproc, rewrites, config, cgutils
     18 from numba.core.typing.templates import (signature, infer_global,

/usr/local/lib/python3.7/dist-packages/numba/core/extending.py in <module>
     17 from numba.core.datamodel import models   # noqa: F401
     18 from numba.core.datamodel import register_default as register_model  # noqa: F401, E501
---> 19 from numba.core.pythonapi import box, unbox, reflect, NativeValue  # noqa: F401
     20 from numba._helperlib import _import_cython_function  # noqa: F401
     21 from numba.core.serialize import ReduceMixin

/usr/local/lib/python3.7/dist-packages/numba/core/pythonapi.py in <module>
     10
     11 import ctypes
---> 12 from numba import _helperlib
     13 from numba.core import (
     14     types, utils, config, lowering, cgutils, imputils, serialize,

ImportError: numpy.core.multiarray failed to import
```

**Comment by willdjennings** [comment]
Had the same issue -- it was fixed by explicitly pip installing `numba==0.53.1` before installing shap. The `numba` module was updated to [0.54.0 as of a few days ago](https://pypi.org/project/numba/#history). And `numba==0.54.0` has [stricter numpy requirement (`numpy>=1.17`)](https://github.com/numba/numba/blob/e031b70c31c8553521ee89b02db8d2d6e995f9eb/setup.py#L25) than the [old requirement in `numba=0.53.1`](https://github.com/numba/numba/blob/97fe221b3704bd17567b57ea47f4fc6604476cf9/setup.py#L25) (`numpy>=1.15`). 

Note that the [setup.py in `shap` ](https://github.com/slundberg/shap/blob/89855e03d7e92ba0d2ec463c0505878f718664e3/setup.py#L228) lists the plain `numba` (without a version) in the `install_requires`. So `pip install shap` will always install the most recent `numba`. This means that shap relies on numba to stay compatible with shap, whenever new release versions are cut which (in this case) failed to be true.

**Comment by windisch** [comment]
Another problem might be the open `numpy` constraint: It seems that `numpy=1.20.0` in combination with `numba==0.54.0` make shap in its latest version work again. This, however, conflicts with TensorFlow, which requires `numpy<1.20.0`. 

Thus, in a fix, I would suggest to not only add an upper bound for the `numba` version, but also for `numpy`.

**Comment by connortann** [comment]
This legacy issue issue should be fixed now, as shap is tested against numba `0.54.0`
